File: examples/simple.js
const Template = imports.gi.Template;
const Gio = imports.gi.Gio;

// Get our file to process
let file = Gio.File.new_for_path("simple.tmpl");

// Create a new template and parse our input file
let tmpl = new Template.Template();
tmpl.parse_file(file, null);

// Create scope for expansion
let scope = Template.Scope.new ();

// Create and assign "title" variable in scope
let title = scope.get("title");
title.assign_string("Example Title");

// Write to stdout
let stream = Gio.UnixOutputStream.new (0, false);

// Expand the template into stream
let expanded = tmpl.expand_string(scope);
log(expanded);

File: examples/simple.py
#!/usr/bin/env python3

import gi

gi.require_version('Template', '1.0')

from gi.repository import Gio
from gi.repository import Template

# Get our file to process
file = Gio.File.new_for_path('simple.tmpl')

# Create a new template and parse our input file
tmpl = Template.Template()
tmpl.parse_file(file, None)

# Create scope for expansion
scope = Template.Scope.new()

# Create and assign "title" variable in scope
title = scope.get('title')
title.assign_string('Example Title')

# Expand the template into stream
expanded = tmpl.expand_string(scope)
print(expanded)
